shaded/client-hadoop3/pom.xml

```diff
@@ -0,0 +1,348 @@
+<!--
+
+    The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+    (the "License"). You may not use this work except in compliance with the License, which is
+    available at www.apache.org/licenses/LICENSE-2.0
+
+    This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+    either express or implied, as more fully set forth in the License.
+
+    See the NOTICE file distributed with this work for information regarding copyright ownership.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>alluxio-shaded</artifactId>
+    <groupId>org.alluxio</groupId>
+    <version>2.9.0-SNAPSHOT</version>
+  </parent>
+  <artifactId>alluxio-shaded-client-hadoop3</artifactId>
```

Comment:
@LuQQiu any idea about this?

Reply:
@Xenorith what do you think about maintaining 2 client jars? I feel like this is a legit request. This one looks like the first time we really start handling those APIs hadoop 3 is not compatible with hadoop 2. And this incompatibility will only get worse while hadoop 3 keeps moving forward.

Reply:
I just think that maybe the user will only need one client jar in running time, the Hadoop 2 or the Hadoop 3. Otherwise, there will be conflict packages most of the time. So I suggest we can choose what you need in the build.

Reply:
Agree that can we use the shaded hadoop module?
if the shaded-Hadoop module needs an extra package, feel free to add them directly to existing shaded hadoop maven module.

Reply:
For how to deal with the backward compatibility or incompatibility between hadoop 2 and 3, guess it's more a question for product @HelloHorizon @apc999

Reply:
For whoever's involved in this discussion, @Xenorith and I have internally discussed on this and we will figure out a way to build both Alluxio client for Hadoop 2 and 3. We can build both and put the two in a directory, then just create a soft link to the default one and put on the classpath.

With Hadoop 3 having more incompatibilities with hadoop 2, we will have to support both and this is a legit request.